src/context/AuthContext.js

```diff
@@ -0,0 +1,36 @@
+import React, { createContext, useState } from "react";
+import { useNavigate } from "react-router-dom";
+
+export const AuthContext = createContext(null);
+
+function AuthContextProvider({children}) {
+  const [isAuth, toggleIsAuth] = useState({isAuth: false, user: ''});
+  const navigate = useNavigate();
+  
+  function logIn(user) {
+    // e.preventDefault();
+    toggleIsAuth({isAuth: true, user: user});
+    console.log("Gebruiker is ingelogd!");
+    navigate('/profile');
+  }
+
+  function logOut(e) {
+    e.preventDefault();
+    toggleIsAuth({isAuth: false, user: ''});
+    console.log("Gebruiker is uitgelogd!");
+    navigate('/');
+  }
+
+  return (
+    <AuthContext.Provider value={{
```

Comment:
Op deze manier is je object niet zo herbruikbaar. Ik weet eigenlijk niet waarom je 'm zou willen hergebruiken haha, en dit werkt gewoon (dat je direct je object als value meegeeft ipv 'm eerst als const te declareren), maar m'n gevoel zegt dat je deze eerst als const een naam wil geven om hem elders aan te kunnen spreken.